Nasal/engine.nas

```diff
@@ -88,6 +88,46 @@ var primerTimer = maketimer(5, func {
     primerTimer.stop();
 });
 
+# ========== oil consumption ======================
+
+var oil_consumption = maketimer(1.0, func {
+    var oil_level = getprop("/engines/active-engine/oil-level");
+    var rpm = getprop("/engines/active-engine/rpm");
+    # quadratic formula which outputs 1.0 for input 2300 RPM (cruise value), 0.6 for 700 RPM (idle) and 1.2 for 2700 RPM (max)
+    var rpm_factor = 0.00000012 * math.pow(rpm,2) - 0.0001 * rpm + 0.62;
+    # consumption rate defined as 1.5 quarter per 10 hours (36000 seconds) at cruise RPM
+    var consumption_rate = 1.5 / 36000; 
+    var low_oil_pressure_factor = 1.0;
+    var low_oil_temperature_factor = 1.0;
+    
+    if ((getprop("/engines/active-engine/running")) and (getprop("/engines/active-engine/oil_consumption_allowed"))) {
+        oil_level = oil_level - consumption_rate * rpm_factor;
+        setprop("/engines/active-engine/oil-level", oil_level);        
+    }
+    
+    # if oil gets low, pressure should lower and temperature should rise
+    if (oil_level < 5.0) {
+        low_oil_pressure_factor = 11.25 * oil_level - 55.25; # should give 1.0 for oil_level = 5 and 0.1 for oil_level 4.92, which is the min before the engine stops
+        low_oil_temperature_factor = -6.25 * oil_level + 32.25; # should give 1.0 for oil_level = 5 and 1.5 for oil_level 4.92
+        setprop("/engines/active-engine/low-oil-pressure-factor", low_oil_pressure_factor);
+        setprop("/engines/active-engine/low-oil-temperature-factor", low_oil_temperature_factor);
+    } else {
+        setprop("/engines/active-engine/low-oil-pressure-factor", 1.0);
+        setprop("/engines/active-engine/low-oil-temperature-factor", 1.0);
+    };
+
+});
+
+# ========== engine coughing ======================
+
+var engine_coughing = maketimer(3.0, func {
+    var coughing = getprop("/engines/active-engine/coughing");
+    var running = getprop("/engines/active-engine/running");
+    if (coughing and running) {
+        var delay = 3.0 * rand();
+        settimer(func {setprop("/engines/active-engine/killed", 1);}, delay); # the <logic> block in engine.xml which controls the engine killed status will force it back to killed == 0 if oil level is still all right
```

Comment:
Move the comment above the code. Make it a multi-line comment if the comment is longer than 72 characters (including the whitespace in front of the comment)